src/Twig/Extension/LanguageSwitcherExtension.php

```diff
@@ -71,12 +101,23 @@ public function getLocalizedLinks(Document $document): array
             }
 
             $route = $request->get('_route');
-            $routeParams = $request->get('_route_params');
 
-            if ($route && $routeParams) {
-                $routeParams['_locale'] = \Locale::getPrimaryLanguage($language);
-                $route = $this->urlGenerator->generate($route, $routeParams);
-                $target = $route;
+            if ($route && array_key_exists($route, $dynamicRoutesMapping)) {
+                $routeParams = $request->get('_route_params', []);
+                $requiredField = $dynamicRoutesMapping[$route]['requiredField'];
+
+                if (!array_key_exists($requiredField, $routeParams)){
+                    continue;
+                }
+
+                $generator = $dynamicRoutesMapping[$route]['generator'];
+                $object = $request->get($requiredField);
+
+                if (!is_object($object)) {
+                    $object = DataObject::getById($object);
+                }
+                $linkGeneratorService = $this->locator->get($generator);
```

Comment:
Isn't it sufficient to pass locale to the generator for generating dynamic routes?

Reply:
like https://github.com/pimcore/demo/pull/481 ? it seems that there are some other parts that are passed as is before any localization eg
https://github.com/pimcore/demo/pull/524/files?w=1#diff-d077cf6e8d598bf4dd034a28ccd7776e15292fe570d69fd3b2b4a216633a489cR101
https://github.com/pimcore/demo/pull/524/files?w=1#diff-5b85377fb245d8c17829306af8de9dbfc4bff5e14597b574b7fda07077e630fbR76

Reply:
maybe it didn't work as the link generators looks for locale with param `locale` and we passed `_locale` here https://github.com/pimcore/demo/pull/524/files#diff-758b76db0f9bd9bdba6e2df9d4b04e58a207c9c2e83f8220ea50948ec81c7a93L77